### PR TITLE
Improve instructions and tests for Remote Control Competition exercise

### PR DESCRIPTION
**Pull request by eneuschild**
# Improve instructions and tests for Remote Control Competition exercise

Issue #2164 points out that Remote Control Competition's instructions don't match its tests very well. This PR is an attempt to improve this.

Completing each instruction subsection now increases the number of passing tests, which wasn't given previously.

## Open Questions
- I have not yet touched the tests themselves, as I feel this would be too big a change for a single (first) PR. Would you agree?
  - **Update**: Since @SleeplessByte disagreed with this assessment, I have now introduced more tests that gradually guide the student to a complete solution. 
- The way I understand the [Updating the README](https://github.com/exercism/java/blob/6ef04c4cd2ab62bca7f8d821ff18c5e81390eedd/CONTRIBUTING.md#updating-the-readmes) section of the CONTRIBUTING.md, I should be running `configlet` to maintain README files. Am I understanding correctly that this isn't really applicable here because this exercise is not part of the more generic problem specifications? :thinking: 

---

Reviewer Resources:

[Track Policies](https://github.com/exercism/java/blob/main/POLICIES.md#event-checklist)

### CODE REVIEW

**Comment by eneuschild** [comment]
Hi @kytrinyx, any advice on how to get this to the next step? 🙂

**Comment by kytrinyx** [comment]
@eneuschild I recommend starting a conversation on the forum: https://forum.exercism.org/

In the past few months, most successful contributions happen by first discussing on the forum, in order to get clarity on what, exactly, the problem is, as well as how to approach resolving it. Once that is done, opening a PR is more of a formality.

**Comment by kytrinyx** [comment]
@eneuschild I just saw your post in the forum (https://forum.exercism.org/t/getting-pr-to-java-track-reviewed/4913). I'm not a track maintainer, so I'm not the right person to review this.

@andrerfcsantos Do you know what the status is in the Java track these days?

**Comment by eneuschild** [comment]
Thanks for the pointers, @kytrinyx! 💐 I'll try to stick to that process in the future! 🙂

**Review by SleeplessByte — APPROVED** [review]
The changes here look sensible. I would recommend to also make the tests changes in THIS PR, because otherwise they become unrelated changes.

**Comment by eneuschild** [comment]
@SleeplessByte Thanks for your feedback!
I have now added some more tests that allow a more gradual approach to a working all-green solution.

Please have another look! :slightly_smiling_face:

File: exercises/concept/remote-control-competition/.docs/instructions.md [hunk 1]
```diff
@@ -6,42 +6,58 @@ An experimental car has been developed and the test track needs to be adapted to
...
+Please add two methods to the `RemoteControlCar` interface:
+- `drive()`, returning nothing, and
+- `getDistanceTravelled()`, returning an `int`.
 
-```java
-TestTrack.race(new ProductionRemoteControlCar());
-TestTrack.race(new ExperimentalRemoteControlCar());
-// this should execute without an exception being thrown
-```
+Then make `ProductionRemoteControlCar` and `ExperimentalRemoteControlCar` implement the `RemoteControlCar` interface. This includes implementing all methods required by the interface.
```
Comment (SleeplessByte) [inline exercises/concept/remote-control-competition/.docs/instructions.md:15]:
```suggestion
Then make `ProductionRemoteControlCar` and `ExperimentalRemoteControlCar` implement the `RemoteControlCar` interface.
This includes implementing all methods required by the interface. 
```

File: exercises/concept/remote-control-competition/.docs/instructions.md [hunk 2]
```diff
@@ -6,42 +6,58 @@ An experimental car has been developed and the test track needs to be adapted to
...
+```java
+TestTrack.race(new ProductionRemoteControlCar());
+TestTrack.race(new ExperimentalRemoteControlCar());
+// this should execute without an exception being thrown
+```
+
+## 4. Allow the production cars to be sorted
 
-Please implement the `Comparable<T>` interface in the `ProductionRemoteControlCar` class. The default sort order for cars should be ascending order of victories.
+Please implement the `Comparable<T>` interface in the `ProductionRemoteControlCar` class. The default sort order for cars should be descending order of victories.
```
Comment (SleeplessByte) [inline exercises/concept/remote-control-competition/.docs/instructions.md:48]:
```suggestion
Please implement the `Comparable<T>` interface in the `ProductionRemoteControlCar` class.
The default sort order for cars should be descending order of victories.
```

**Review by SleeplessByte — APPROVED** [review]
Looks good.

**Pull request merged**